core-java-modules/core-java-strings/src/test/java/com/baeldung/randomstrings/RandomStringsUnitTest.java

```diff
@@ -1,72 +1,101 @@
-        final String generatedString = RandomStringUtils.randomAlphanumeric(10);
-
-        LOG.debug(generatedString);
-    }
-
-    @Test
-    public void givenUsingApache_whenGeneratingRandomStringBounded_thenCorrect() {
-        final int length = 10;
-        final boolean useLetters = true;
-        final boolean useNumbers = false;
-        final String generatedString = RandomStringUtils.random(length, useLetters, useNumbers);
-
-        LOG.debug(generatedString);
-    }
-
-}
+package com.baeldung.randomstrings;
+
+import org.apache.commons.lang3.RandomStringUtils;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.nio.charset.Charset;
+import java.util.Random;
+
+public class RandomStringsUnitTest {
+
+    private static final Logger LOG = LoggerFactory.getLogger(RandomStringsUnitTest.class);
+
+    @Test
+    public void givenUsingPlainJava_whenGeneratingRandomStringUnbounded_thenCorrect() {
+        final byte[] array = new byte[7]; // length is bounded by 7
+        new Random().nextBytes(array);
+        final String generatedString = new String(array, Charset.forName("UTF-8"));
+
+        LOG.debug(generatedString);
+    }
+
+    @Test
+    public void givenUsingPlainJava_whenGeneratingRandomStringBounded_thenCorrect() {
+        final int leftLimit = 97; // letter 'a'
+        final int rightLimit = 122; // letter 'z'
+        final int targetStringLength = 10;
+        final Random random = new Random();
+        final StringBuilder buffer = new StringBuilder(targetStringLength);
+
+        for (int i = 0; i < targetStringLength; i++) {
+            final int randomLimitedInt = leftLimit + (int) (random.nextFloat() * (rightLimit - leftLimit + 1));
+            buffer.append((char) randomLimitedInt);
+        }
+        final String generatedString = buffer.toString();
+
+        LOG.debug(generatedString);
+    }
+
+    @Test
+    public void givenUsingJava8_whenGeneratingRandomAlphabeticString_thenCorrect() {
+        final int leftLimit = 97; // letter 'a'
+        final int rightLimit = 122; // letter 'z'
+        final int targetStringLength = 10;
+        final Random random = new Random();
+
+        final String generatedString = random.ints(targetStringLength, leftLimit, rightLimit + 1)
+            .collect(StringBuilder::new, StringBuilder::appendCodePoint, StringBuilder::append)
+            .toString();
+
+        LOG.debug(generatedString);
+    }
+
+    @Test
+    public void givenUsingJava8_whenGeneratingRandomAlphanumericString_thenCorrect() {
+        final int leftLimit = 48; // numeral '0'
+        final int rightLimit = 122; // letter 'z'
+        final int targetStringLength = 10;
+        final Random random = new Random();
+
+        final String generatedString = random.ints(targetStringLength, leftLimit, rightLimit + 1)
+            .filter(i -> (i < 57 || i > 65) && (i < 90 || i > 97))

```

Comment:
Sorry to bother you, but the length of the result string varies because of this filter, I doubt that that is desirable.

Maybe add assertions (about the length) to both tests.

And add a limit after the filter instead of before.

PS: I'm also surprised to see `Random `being used here instead of `SecureRandom`

Reply:
And one more it seems your filter also excludes some wrong values, maybe you better extract it as a separate method and do some boundary tests on it.

The following values are filtered out: 9:;<=>?@AZ[\]^_`a

Of which 9, A, Z and a don't look like values you would like to filter out.

Reply:
Thanks @martinvw for the suggested improvements, i have incorporated them. For the Random vs SecureRandom thing, please refer to the JIRA ticket's description! :)